Lib/typing.py

```diff
@@ -164,14 +164,16 @@ def _type_check(arg, msg, is_argument=True, module=None):
     We append the repr() of the actual value (truncated to 100 chars).
     """
     invalid_generic_forms = (Generic, Protocol)
-    if is_argument:
-        invalid_generic_forms = invalid_generic_forms + (ClassVar, Final)
+    if not is_class:
+        invalid_generic_forms += (ClassVar,)
+        if is_argument:
+            invalid_generic_forms += (Final,)
```

Comment:
Note to self: there's a subtle change in the logic compared to the old code, but it's ok because ``is_argument`` defaulted to ``True`` in both `ForwardRef` and `_type_check`. Which means by default we should always add ``ClassVar`` too.